generator/views.py

```diff
@@ -91,11 +119,289 @@ def from_db_views(
         for view_id, tables in views.items():
             yield PingView(view_id, tables)
 
+    @classmethod
+    def from_dict(klass, name: str, _dict: ViewDict) -> PingView:
+        """Get a view from a name and dict definition."""
+        return PingView(name, _dict["tables"])
+
+    def to_lookml(self, bq_client) -> List[dict]:
+        """Generate LookML for this view."""
+        view_defn: Dict[str, Any] = {"name": self.name}
+
+        # use schema for the table where channel=="release" or the first one
+        table = next(
+            (table for table in self.tables if table.get("channel") == "release"),
+            self.tables[0],
+        )["table"]
+
+        # add dimensions and dimension groups
+        dimensions = lookml_utils._generate_dimensions(bq_client, table)
+        view_defn["dimensions"] = list(
+            filterfalse(lookml_utils._is_dimension_group, dimensions)
+        )
+        view_defn["dimension_groups"] = list(
+            filter(lookml_utils._is_dimension_group, dimensions)
+        )
+
+        # add measures
+        view_defn["measures"] = self.get_measures(dimensions, table)
+
+        # parameterize table name
+        if len(self.tables) > 1:
+            view_defn["parameters"] = [
+                {
+                    "name": "channel",
+                    "type": "unquoted",
+                    "allowed_values": [
+                        {
+                            "label": table["channel"].title(),
+                            "value": table["table"],
+                        }
+                        for table in self.tables
+                    ],
+                }
+            ]
+            view_defn["sql_table_name"] = "`{% parameter channel %}`"
+        else:
+            view_defn["sql_table_name"] = f"`{table}`"
+
+        return [view_defn]
+
+    def get_measures(self, dimensions: List[dict], table: str) -> List[Dict[str, str]]:
+        """Generate measures from a list of dimensions.
+
+        When no dimension-specific measures are found, return a single "count" measure.
+
+        Raise ClickException if dimensions result in duplicate measures.
+        """
+        measures = {}
+
+        for dimension in dimensions:
+            dimension_name = dimension["name"]
+            if dimension_name in {"client_id", "client_info__client_id"}:
+                measure = {
+                    "name": "clients",
+                    "type": "count_distinct",
+                    "sql": f"${{{dimension_name}}}",
+                }
+            elif dimension_name == "document_id":
+                measure = {"name": "ping_count", "type": "count"}
```

Comment:
It should be fine to keep this as a `count`, since we don't have any joins yet. Once we enable joins we can certainly add `document_id` as the primary key.